js/ccf-app/src/polyfill.ts

```diff
@@ -381,40 +380,98 @@ class CCFPolyfill implements CCF {
       }
     },
     pubPemToJwk(pem: string, kid?: string): JsonWebKeyECPublic {
-      let jwk = rs.KEYUTIL.getJWK(
-        rs.KEYUTIL.getKey(pem) as rs.KJUR.crypto.ECDSA
-      ) as JsonWebKeyECPublic;
-      if (kid !== undefined) {
-        jwk.kid = kid;
-      }
-      return jwk;
+      const key = jscrypto.createPublicKey({
+        key: pem,
+      });
+      const jwk = key.export({
+        format: "jwk",
+      });
+      return {
+        crv: jwk.crv as string,
+        x: jwk.x as string,
+        y: jwk.y as string,
+        kty: jwk.kty as string,
+        kid: kid,
+      };
     },
     pemToJwk(pem: string, kid?: string): JsonWebKeyECPrivate {
-      let jwk = rs.KEYUTIL.getJWK(
-        rs.KEYUTIL.getKey(pem) as rs.KJUR.crypto.ECDSA
-      ) as JsonWebKeyECPrivate;
-      if (kid !== undefined) {
-        jwk.kid = kid;
-      }
-      return jwk;
+      const key = jscrypto.createPrivateKey({
+        key: pem,
+      });
+      const jwk = key.export({
+        format: "jwk",
+      });
+      return {
+        d: jwk.d as string,
+        crv: jwk.crv as string,
+        x: jwk.x as string,
+        y: jwk.y as string,
+        kty: jwk.kty as string,
+        kid: kid,
+      };
```

Comment:
I'm a little worried about this, as we manually construct the object rather than just using `jwk` as created on `:401`. Is there a reason for this? I'm thinking that if `jwk` had an extra field `foo` in it, we wouldn't return it here.

Reply:
If we are developing a general crypto library, we shouldn't miss `foo`, but it's polyfill which should simulate the CCF behaviour. Considering non-polyfill response is strongly typed by C++ struct and can't handle `foo`, I think it's OK.

Reply:
As long as our implementation is tested separately against a well-known library (that's what the Python test is here for), then I think we're good.